contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/ForeignMultiAMBErc20ToErc677.sol

```diff
@@ -201,7 +201,11 @@ contract ForeignMultiAMBErc20ToErc677 is BasicMultiAMBErc20ToErc677 {
     * @param _token address of the token contract.
     * @param _receiver the address that will receive the tokens on the other network.
     */
-    function fixMediatorBalance(address _token, address _receiver) public onlyIfUpgradeabilityOwner {
+    function fixMediatorBalance(address _token, address _receiver)
+        external
+        onlyIfUpgradeabilityOwner
+        validAddress(_receiver)
+    {
         require(isTokenRegistered(_token));
         uint256 balance = ERC677(_token).balanceOf(address(this));
         uint256 expectedBalance = mediatorBalance(_token);
```

Comment:
Should `mediatorBalance` be used from MediatorBalanceStorage.sol? It will require updating of `multi_amb_erc20_to_erc677/callflows.md`

Reply:
No, we cannot use `mediatorBalance` from the `MediatorBalanceStorage`, since its signature is `mediatorBalance()`, not `mediatorBalance(address)`. So, this contract defines and uses its own versions of `mediatorBalance`/`_setMediatorBalance` functions.

Reply:
Will it increase the runtime byte code size if `mediatorBalance(address)` and `_setMediatorBalance(address)` are part of the MediatorBalanceStorage.sol?

Reply:
Yes, it will increase the runtime byte code size for sure for all of the mediators contracts. I am not sure if a compiler can remove unused internal functions, but for sure it can't decide to remove public or external functions.

I have created a `MediatorBalanceStorage.sol` to extract the duplicated methods from regular mediators to a single file, but for multi token extension we don't need to do so, since we have only one such contract for now.

If it will be necessary in the future, we can create a `MediatorMultiBalanceStorage.sol` in order to keep  `mediatorBalance(address)` and `_setMediatorBalance(address)` there.

Reply:
> for sure it can't decide to remove public or external functions.

it makes sense